### ISSUE TITLE
Should a `new` expression with type arguments need parentheses?

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#37523]
```ts
// @target es6

// @filename: file1.ts
let stuff1 = new Array<string>;

// @filename: file2.ts
let stuff2 = new Map<string, number>;
```

This code currently errors with

```
'(' expected.
```

I actually don't care if this code should be allowed or not.

Maybe it should, in which case this is a bug on needing to allow the code.

Maybe it shouldn't, in which case I think we should give a better error message like

```
A 'new' expression with type arguments must always be followed by a parenthesized argument list.
```

**Comment by Rustin-Liu** [comment]
I want to working on it, can I take this?   Could you please add more info about how to fix it?

**Comment by DanielRosenwasser** [comment]
1. Add the new error message to `src/compiler/diagnosticMessages.json`
1. Run `npx gulp generate-diagnostics`
1. In `src/compiler/parser.ts`

   https://github.com/microsoft/TypeScript/blob/37569d01f6a67ef4cca4fac46dfbff646137b9d9/src/compiler/parser.ts#L5202

   you'll want to write something like

   ```ts
   if (token() === SyntaxKind.OpenParenToken) {
       node.arguments = parseArgumentList();
   }
   else if (node.typeArguments) {
       // issue the error here
   }
   ```
1. Try running `npx gulp runtests-parallel`. There is at least one test that should break:

   https://github.com/microsoft/TypeScript/blob/37569d01f6a67ef4cca4fac46dfbff646137b9d9/tests/cases/conformance/expressions/newOperator/newOperatorErrorCases.ts#L32

1. Run `npx gulp baseline-accept`

**Comment by DanielRosenwasser** [comment]
Thanks @Rustin-Liu!
